cli/azd/internal/scaffold/scaffold.go

```diff
@@ -115,35 +116,94 @@ func ExecInfra(
 		return err
 	}
 
-	if spec.DbCosmosMongo != nil {
-		err = Execute(t, "db-cosmos-mongo.bicep", spec.DbCosmosMongo, filepath.Join(infraApp, "db-cosmos-mongo.bicep"))
-		if err != nil {
-			return fmt.Errorf("scaffolding cosmos mongodb: %w", err)
-		}
+	err = Execute(t, "main.bicep", spec, filepath.Join(infraRoot, "main.bicep"))
+	if err != nil {
+		return fmt.Errorf("scaffolding main.bicep: %w", err)
 	}
 
-	if spec.DbPostgres != nil {
-		err = Execute(t, "db-postgres.bicep", spec.DbPostgres, filepath.Join(infraApp, "db-postgres.bicep"))
+	err = Execute(t, "resources.bicep", spec, filepath.Join(infraRoot, "resources.bicep"))
+	if err != nil {
+		return fmt.Errorf("scaffolding resources.bicep: %w", err)
+	}
+
+	err = Execute(t, "main.parameters.json", spec, filepath.Join(infraRoot, "main.parameters.json"))
+	if err != nil {
+		return fmt.Errorf("scaffolding main.parameters.json: %w", err)
+	}
+
+	return nil
+}
+
+// ExecInfra scaffolds infrastructure files for the given spec, using the loaded templates in t. The resulting files
+// are written to the target directory.
+func ExecInfraFs(
+	t *template.Template,
+	spec InfraSpec) (*memfs.FS, error) {
+	fs := memfs.New()
+
+	// Pre-execution expansion. Additional parameters are added, derived from the initial spec.
+	preExecExpand(&spec)
+
+	err := copyFsToMemFs(resources.ScaffoldBase, fs, baseRoot, ".")
```

Comment:
![image](https://github.com/user-attachments/assets/27207130-3049-4fc6-b3f1-d5ed8306a4ff)

ScaffoldBase is already a memory file system. Can you help me understand why do we need to copy it to a new memfs?

Reply:
`ScaffoldBase` is a filesystem containing assets to generate from.

Here, we are generating an in-memory filesystem contains the actual `infra` Bicep files that would be used for `azd provision`. This is how Aspire works today as well.

Reply:
I'm confused about the copy from fs to a new fs.
We create the template directly from the fs with `ParseFS()` https://github.com/Azure/azure-dev/blob/main/cli/azd/pkg/apphost/generate.go#L99

Then executeToFS() uses an existing fs.   Is this `existing fs` the equivalent to the new fs you are creating?

That's why I wonder if you can just re-use the same fs, maybe just to use a root folder like `out` ?